src/services/bboxService.ts

```diff
@@ -105,7 +105,7 @@ export const getBboxPages = (bboxes: IBboxLocation[], structure: AnyObject) => {
 }
 
 export const checkIsBboxOutOfThePage = (bbox: IBbox, scale: number) => {
-  const parent = (document.querySelector('.pdf-viewer') as any);
+  const parent = (document.querySelector('.react-pdf__Page') as any);
```

Comment:
Looks like it returns the first page only, but we should check the page that contains this bbox. You can pass the bbox pageNumber here and seach using the data-page-number attribute